pkg/clients/postgresql/postgresql.go

```diff
@@ -116,6 +128,7 @@ func (c postgresDB) GetConnectionDetails(username, password string) managed.Conn
 		xpv1.ResourceCredentialsSecretPasswordKey: []byte(password),
 		xpv1.ResourceCredentialsSecretEndpointKey: []byte(c.endpoint),
 		xpv1.ResourceCredentialsSecretPortKey:     []byte(c.port),
+		sslModeKey:                                []byte(c.sslmode),
```

Comment:
Looks like we ended up not reading from connection details in favor of provider config spec. This like should be removed.